src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseStore.cs

```diff
@@ -32,14 +33,17 @@ internal class DocumentServiceLeaseStore : ILeaseStore
             this.containerNamePrefix = containerNamePrefix;
             this.leaseCollectionLink = leaseCollectionLink;
             this.requestOptionsFactory = requestOptionsFactory;
+            this.isPartitionedByLeasePk = this.requestOptionsFactory is PartitionedByLeasePkCollectionRequestOptionsFactory;
         }
 
         public async Task<bool> IsInitializedAsync()
         {
             string markerDocId = this.GetStoreMarkerName();
             Uri documentUri = UriFactory.CreateDocumentUri(this.leaseStoreCollectionInfo.DatabaseName, this.leaseStoreCollectionInfo.CollectionName, markerDocId);
-            var requestOptions = this.requestOptionsFactory.CreateRequestOptions(
-                DocumentServiceLease.FromDocument(new Document { Id = markerDocId }));
+            var documentServiceLease = DocumentServiceLease.FromDocument(new Document { Id = markerDocId });
+            if (this.isPartitionedByLeasePk)
+                documentServiceLease.LeasePartitionKey = markerDocId;
```

Comment:
I don't think any modifications are needed in this method. As we need is uri and request options. We are not creating new doc in lease collection.